### PR TITLE
Feat/header

### PR DESCRIPTION
**Pull request by Romeo1928**
update header

### CODE REVIEW

File: public/locales/en.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "button": "Go back to home page"
   },
   "search": {
-    "searchPlaceholder": "Search..."
+    "searchPlaceholder": "Search friends..."
```
Comment (AdammmVV) [inline public/locales/en.json]: search for users here

File: src/components/shared/search/InputSearch.tsx [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-} from '@mantine/core';
-import { IconSearch, IconArrowRight } from '@tabler/icons-react';
+import { rem, TextInput, TextInputProps } from '@mantine/core';
+import { IconSearch } from '@tabler/icons-react';
 import { useTranslation } from 'react-i18next';
+import { ChangeEvent } from 'react';
 
 export function InputSearch(props: TextInputProps) {
-  const theme = useMantineTheme();
+  // const theme = useMantineTheme();
```
Comment (AdammmVV) [inline src/components/shared/search/InputSearch.tsx]: remove unnecessary comments

File: src/components/shared/search/InputSearch.tsx [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
-      rightSection={
-        <ActionIcon size={32} radius="md" color={theme.primaryColor} variant="filled">
-          <IconArrowRight style={{ width: rem(18), height: rem(18) }} stroke={1.5} />
-        </ActionIcon>
-      }
+      // rightSection={
+      //   <ActionIcon size={32} radius="md" color={theme.primaryColor} variant="filled">
+      //     <IconArrowRight style={{ width: rem(18), height: rem(18) }} stroke={1.5} />
+      //   </ActionIcon>
+      // }
```
Comment (AdammmVV) [inline src/components/shared/search/InputSearch.tsx]: remove unnecessary comments

File: src/layout/header/AppLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { FC } from 'react';
+import { Group, Image, Text } from '@mantine/core';
+import logo from 'assets/logo/shuriken.png';
+import { useNavigate } from 'react-router-dom';
+
+export const AppLogo: FC = () => {
+    const navigate = useNavigate();
+    const handleHomeClick = () => {
+        navigate('/');
```
Comment (AdammmVV) [inline src/layout/header/AppLogo.tsx]: use Enum for pths `'/'`

File: src/layout/header/AppLogo.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { FC } from 'react';
+import { Group, Image, Text } from '@mantine/core';
+import logo from 'assets/logo/shuriken.png';
+import { useNavigate } from 'react-router-dom';
+
+export const AppLogo: FC = () => {
+    const navigate = useNavigate();
+    const handleHomeClick = () => {
```
Comment (AdammmVV) [inline src/layout/header/AppLogo.tsx]: add emty line

File: src/layout/header/AppLogo.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import logo from 'assets/logo/shuriken.png';
+import { useNavigate } from 'react-router-dom';
+
+export const AppLogo: FC = () => {
+    const navigate = useNavigate();
+    const handleHomeClick = () => {
+        navigate('/');
+    };
+    return (
+        <Group style={{ cursor: 'pointer' }} onClick={handleHomeClick} justify="center" grow>
```
Comment (AdammmVV) [inline src/layout/header/AppLogo.tsx]: do not use inline styles

File: src/layout/header/Header.tsx [hunk 7]
```diff
@@ -1,94 +1,118 @@
...
+  // const auth = useAuth();
+  // const dispatch = useAppDispatch();
 
-  useEffect(() => {
-    if (auth.id) {
-      dispatch(loadUserAvatar(auth.id));
-    }
-  }, [dispatch, auth.id]);
+  const user = useAppSelector(getUser);
+  const isAuthenticated = user && !!user.login;
```
Comment (AdammmVV) [inline src/layout/header/Header.tsx]: add emty line

File: src/layout/header/Header.tsx [hunk 8]
```diff
@@ -1,94 +1,118 @@
...
+import { HeaderLoginForm } from 'layout/header/HeaderLoginForm';
+import { useAppSelector } from 'hooks/redux-hooks';
 
 export function HeaderMenu() {
-  const navigate = useNavigate();
-  const { t } = useTranslation();
-  const auth = useAuth();
-  const isAuthenticated = auth && !!auth.login;
-  const dispatch = useAppDispatch();
+  // const navigate = useNavigate();
```
Comment (AdammmVV) [inline src/layout/header/Header.tsx]: remove unnecessary comments and throughout PR

File: src/layout/header/HeaderLoginForm.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { FC } from 'react';
+import { Button } from '@mantine/core';
+import { LOGIN_ROUTE, REGISTRATION_ROUTE } from 'utils/constsRoutes';
+import { useNavigate } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+
+export const HeaderLoginForm: FC = () => {
+    const navigate = useNavigate();
+    const { t } = useTranslation();
+    const handleLoginClick = () => {
```
Comment (AdammmVV) [inline src/layout/header/HeaderLoginForm.tsx]: add emty line

File: src/layout/header/HeaderUserProfile.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    if (user.id) {
+      dispatch(loadUserAvatar(user.id));
+    }
+  }, [dispatch, user.id]);
+
+  const navigate = useNavigate();
+  const handleProfileClick = () => {
+    navigate(PROFILE_ROUTE);
+  };
+
```
Comment (AdammmVV) [inline src/layout/header/HeaderUserProfile.tsx]: place the hooks on top, useEffect above the return.
add empty lines for hooks with different meanings, for example: react, redux, router, custom

**Comment by AdammmVV** [comment]
Check what the project is not building or ask Raf, let him look at the vecel logs, or try to build the project locally
image

File: src/layout/header/AuthenticatedContent/AuthenticatedContent.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+      ) : (
+        <Group>
+          <LanguageSwitcher />
+          <ActionToggleTheme />
+          <HeaderShoppingCart />
+          <HeaderNotify />
+          <HeaderUserProfile />
+        </Group>
+      )}
+    </Group>
```
Comment (AdammmVV) [inline src/layout/header/AuthenticatedContent/AuthenticatedContent.tsx]:
```suggestion
      {isMatches && (
        <AppShell.Navbar p="md">
          <Group>
            <LanguageSwitcher />
            <ActionToggleTheme />
            <HeaderShoppingCart />
            <HeaderNotify />
            <HeaderUserProfile />
          </Group>

          <Divider my="md" />

          <Paper className="navbar">
            <Navbar />
          </Paper>
        </AppShell.Navbar>
      )}
       {!isMatches && (
        <Group>
          <LanguageSwitcher />
          <ActionToggleTheme />
          <HeaderShoppingCart />
          <HeaderNotify />
          <HeaderUserProfile />
        </Group>
      )}
    </Group>
```

File: src/layout/header/Header/Header.module.css [hunk 12]
```diff
@@ -1,10 +1,11 @@
...
-  border-bottom: rem(1px) solid
-    light-dark(var(--mantine-color-gray-3), var(--mantine-color-dark-4));
-}
+/*.header {*/
+/*  height: rem(70px);*/
+/*  padding-left: var(--mantine-spacing-md);*/
+/*  padding-right: var(--mantine-spacing-md);*/
+/*  border-bottom: rem(1px) solid*/
+/*    light-dark(var(--mantine-color-gray-3), var(--mantine-color-dark-4));*/
+/*}*/
```
Comment (AdammmVV) [inline src/layout/header/Header/Header.module.css]: remove unnecessary comments

File: src/layout/navbar/NavBar.tsx [hunk 13]
```diff
@@ -69,7 +69,8 @@ export function Navbar() {
     ));
 
   return (
-    <nav className={classes.navbar}>
+    // <nav className={classes.navbar}>
```
Comment (AdammmVV) [inline src/layout/navbar/NavBar.tsx]: remove unnecessary comments

File: src/redux/slice/userSlice.ts [hunk 14]
```diff
@@ -220,3 +220,5 @@ const authSlice = createSlice({
   },
 });
 export default authSlice.reducer;
+
+export const getUser = (state: RootState) => state.user;
```
Comment (AdammmVV) [inline src/redux/slice/userSlice.ts]: put selectors in a separate file

File: src/layout/header/AppLogo/AppLogo.tsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
+import { FC } from 'react';
+import { HOME_ROUTE } from 'utils/constsRoutes';
+import { useNavigate } from 'react-router-dom';
+import { useMediaQuery } from '@mantine/hooks';
+
+export const AppLogo: FC = () => {
+  const isMatches = useMediaQuery('(max-width: 440px)');
```
Comment (AdammmVV) [inline src/layout/header/AppLogo/AppLogo.tsx:7]: add color for ligth and darck theme

**Comment by alexspel** [comment]
+

**Pull request merged**